patch/kernel/archive/sunxi-6.2/patches.armbian/arm64-dts-sun50i-h616-bigtreetech-cb1.patch

```diff
@@ -0,0 +1,453 @@
+ dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h616-orangepi-zero2.dtb
+ dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h616-x96-mate.dtb
++dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h616-bigtreetech-cb1-sd.dtb
++dtb-$(CONFIG_ARCH_SUNXI) += sun50i-h616-bigtreetech-cb1-emmc.dtb
+ 
+ subdir-y	:= $(dts-dirs) overlay
+diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1-emmc.dts b/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1-emmc.dts
+new file mode 100644
+index 000000000000..f878c23f1d90
+--- /dev/null
++++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1-emmc.dts
+@@ -0,0 +1,44 @@
++// SPDX-License-Identifier: (GPL-2.0+ or MIT)
++/*
++ * Copyright (C) 2023 Alan.Ma <tech@biqu3d.com>.
++ */
++
++/dts-v1/;
++
++#include "sun50i-h616-bigtreetech-cb1.dtsi"
++
++&mmc2 {
++	vmmc-supply = <&reg_dldo1>;
++
++	no-1-8-v;
++	bus-width = <8>;
++	non-removable;
++	status = "okay";
++};
++
++&ws2812 {
++	gpios = <&pio 8 15 GPIO_ACTIVE_LOW>; /* PI15 */
++};
++
++&i2c_gpio {
++	gpios = <&pio 8 6 GPIO_ACTIVE_HIGH>, /* SDA PI6 */
++			<&pio 8 4 GPIO_ACTIVE_HIGH>; /* SCL PI4 */
++};
++
++&can0_pin_irq {
++	pins = "PI3";
++};
++
++&can {
++	interrupts = <8 3 0x08>; /* PI3 IRQ_TYPE_LEVEL_LOW */
++};
++
++&tft_35 {
++	dc-gpios  = <&pio 8 15 GPIO_ACTIVE_HIGH>; /* PI15 */
++};
++
++&spi1 {
++	cs-gpios = <&pio 8 5 GPIO_ACTIVE_HIGH>, /* PI5 */
++			   <&pio 8 14 GPIO_ACTIVE_HIGH>, /* PI14 */
++			   <&pio 8 7 GPIO_ACTIVE_HIGH>; /* PI7 */
++};
+diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1-sd.dts b/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1-sd.dts
+new file mode 100644
+index 000000000000..e18dd854d74b
+--- /dev/null
++++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1-sd.dts
+@@ -0,0 +1,35 @@
++// SPDX-License-Identifier: (GPL-2.0+ or MIT)
++/*
++ * Copyright (C) 2023 Alan.Ma <tech@biqu3d.com>.
++ */
++
++/dts-v1/;
++
++#include "sun50i-h616-bigtreetech-cb1.dtsi"
++
++&ws2812 {
++	gpios = <&pio 2 14 GPIO_ACTIVE_LOW>; /* PC14 */
++};
++
++&i2c_gpio {
++	gpios = <&pio 2 12 GPIO_ACTIVE_HIGH>, /* SDA PC12 */
++			<&pio 2 10 GPIO_ACTIVE_HIGH>; /* SCL PC10 */
++};
++
++&can0_pin_irq {
++	pins = "PC9";
++};
++
++&can {
++	interrupts = <2 9 0x08>; /* PC9 IRQ_TYPE_LEVEL_LOW */
++};
++
++&tft_35 {
++	dc-gpios  = <&pio 2 14 GPIO_ACTIVE_HIGH>; /* PC14 */
++};
++
++&spi1 {
++	cs-gpios = <&pio 2 11 GPIO_ACTIVE_HIGH>, /* PC11 */
++			   <&pio 2 7 GPIO_ACTIVE_HIGH>, /* PC7 */
++			   <&pio 2 13 GPIO_ACTIVE_HIGH>; /* PC13 */
++};
+diff --git a/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1.dtsi b/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1.dtsi
+new file mode 100644
+index 000000000000..c80cc830c321
+--- /dev/null
++++ b/arch/arm64/boot/dts/allwinner/sun50i-h616-bigtreetech-cb1.dtsi
+@@ -0,0 +1,327 @@
++// SPDX-License-Identifier: (GPL-2.0+ or MIT)
++/*
++ * Copyright (C) 2023 Alan.Ma <tech@biqu3d.com>.
++ */
++
++/dts-v1/;
++
++#include "sun50i-h616.dtsi"
++#include "sun50i-h616-cpu-opp.dtsi"
```

Comment:
The patches in 6.1 and 6.2 are exactly the same except for this line.
6.1 has not  `#include "sun50i-h616-cpu-opp.dtsi"`

Reply:
> Please point me to a copy of the opp.dtsi you are referencing?

Hi, Same file as orangepizero2 here
https://github.com/armbian/build/blob/main/patch/kernel/archive/sunxi-6.2/patches.armbian/arm64-dts-allwinner-h616-Add-efuse_xlate-cpu-frequency-scaling-v1_6_2.patch#L17

Reply:
Thanks for clarifying, in this case you will have to follow the xlate patch conventions including the file changes.

Reply:
CHANGES: required here - drivers/cpufreq/sun50i-cpufreq-nvmem.c  - for example

https://github.com/bigtreetech/linux/blob/0d85b27b0cc6b5cf54567c5ad913a247a71583ce/drivers/cpufreq/sun50i-cpufreq-nvmem.c#L23-L24

Reply:
Here example option for v2 including variants (disclaimer not well tested, but you can try). Further testing is being done with PLL_CPUX.

```
// SPDX-License-Identifier: (GPL-2.0+ OR MIT)
// Copyright (C) 2022 Alistair Musson <AGM1968@users.noreply.github.com>
/* 
ORANGEPIZERO2

 opp_l_table:	bin0  	variant		bin1 	variant
 480000000  	820mV	(0003)		880mV	(0004) 	 
 600000000  	820mV  	(0003)		880mV  	(0004)
 792000000  	860mV	(0003)		940mV  	(0004)
 1008000000 	900mV  	(0003)		1020mV 	(0004)
 1200000000 	960mV  	(0003)		1100mV 	(0004)
 1296000000 	1100mV 	(0002) //allow for variant #2
 1344000000 	1120mV	(0004) //allow for variant #4
 1512000000 	1100mV 	(0001)  

A single regulator's voltage is specified with an array of size one or three.
Single entry is for target voltage and three entries are for <target min max> voltages. 
*/

/ {
	cpu_opp_table: cpu-opp-table {
		compatible = "operating-points-v2";
		opp-shared;

		opp-480000000 {
			clock-latency-ns = <244144>; /* 8 32k periods */
			opp-hz = /bits/ 64 <480000000>;
			opp-microvolt = <820000 820000 880000>;
		};

		opp-600000000 {
			clock-latency-ns = <244144>; /* 8 32k periods */
			opp-hz = /bits/ 64 <600000000>;
			opp-microvolt = <820000 820000 880000>;
		};

		opp-792000000 {
			clock-latency-ns = <244144>; /* 8 32k periods */
			opp-hz = /bits/ 64 <792000000>;
			opp-microvolt = <860000 860000 940000>;
		};

		opp-1008000000 {
			clock-latency-ns = <244144>; /* 8 32k periods */
			opp-hz = /bits/ 64 <1008000000>;
			opp-microvolt = <900000 900000 1020000>;
		};

		opp-1200000000 {
			clock-latency-ns = <244144>; /* 8 32k periods */
			opp-hz = /bits/ 64 <1200000000>;
			opp-microvolt = <960000 960000 1100000>;
		};		

		opp-1512000000 {
			clock-latency-ns = <244144>; /* 8 32k periods */
			opp-hz = /bits/ 64 <1512000000>;
			opp-microvolt = <1100000 1100000 1100000>;
		};		
	};
};

&cpu0 {
	operating-points-v2 = <&cpu_opp_table>;
};

&cpu1 {
	operating-points-v2 = <&cpu_opp_table>;
};

&cpu2 {
	operating-points-v2 = <&cpu_opp_table>;
};

&cpu3 {
	operating-points-v2 = <&cpu_opp_table>;
};
```